packages/instantsearch.js/src/components/Template/Template.tsx

```diff
@@ -85,6 +98,11 @@ See: https://www.algolia.com/doc/guides/building-search-ui/upgrade-guides/js/#up
       return <RootTagName {...this.props.rootProps}>{content}</RootTagName>;
     }
 
+    // This is to handle Hogan templates with Fragment as rootTagName
+    if (RootTagName === Fragment) {
+      return <RawHtml content={content} />;
+    }
```

Comment:
Should this be also conditioned by `nonFunctionTemplates` (which would need to be hoisted one level up in that case)? I think we could just let `RootTagName` be `Fragment` with html templating.

Reply:
Isn't that because Fragment can't have dangerousInnerHTML as in the description? It's already conditioned to be only strings though, as right before it's listed as not being an object.

Simple strings now fall into this case but that's probably ok?

Reply:
Ah well that's it, and it was already mentioned in the description. Thanks for the clarification!

Reply:
Ah yes cause we allow functions to return strings as well, although those strings should not contain HTML.